src/DataCollector/DecoratingFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of php-cache\cache-bundle package.
+ *
+ * (c) 2015-2015 Aaron Scherer <aequasi@gmail.com>, Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Cache\CacheBundle\DataCollector;
+
+use Nyholm\NSA;
+
+/**
+ * A factory that decorates another factory to be able to use the proxy cache.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class DecoratingFactory
+{
+    /**
+     * @type ProxyFactory
+     */
+    private $proxyFactory;
+
+    /**
+     * @type mixed cache pool
+     */
+    private $originalObject;
+
+    /**
+     * @param ProxyFactory $proxyFactory
+     * @param mixed        $originalObject
+     */
+    public function __construct(ProxyFactory $proxyFactory, $originalObject)
+    {
+        $this->proxyFactory   = $proxyFactory;
+        $this->originalObject = $originalObject;
+    }
+
+    public function create()
```

Comment:
Use $originalObject as a function parameter and register the factory in the service.yml